client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -400,7 +409,12 @@ const mapDispatchToProps = {
 	trashPost,
 };
 
-export default connect(
-	mapStateToProps,
-	mapDispatchToProps
-)( CalypsoifyIframe );
+const enhance = flow(
+	protectForm,
+	connect(
+		mapStateToProps,
+		mapDispatchToProps
+	)
+);
```

Comment:
I probably would have wrapped `CalypsoifyIFrame` in `protectForm` instead of using this composition here after reading about the props set by `protectForm`. Right now it's like we can use a connected version of the component that isn't protected yet we rely on those props being there.

may not be a big deal, but in some cases, such as testing, it could end up being a bit confusing.

any thoughts?

Reply:
Don't we use the same pattern with the connected properties? I mean, it also looks like we can have an unconnected version of the component which obviously won't work because the component relies on these properties coming from state and actions. That's the reason why I declared the properties on `PropTypes`, to make explicit the component needs them.

Reply:
it may just be that it's rare enough to see connected components that use the `protectForm` enhancer.

we intentionally create an _unwrapped_ and a _wrapped_ version of components with `connect()` because it makes for simpler testing and it can help clarify what the expectations are for the component. for example, we can define the API for the component and know that it needs something like a `siteId` - `connect()` can provide that but what matters is that it wants a `siteId`, not that it, oh, needs `getSelectedSiteId()` which is irrelevant to the component.

in the case of `protectForm()` we're a bit more tightly-coupled because `markChanged()` and `markSaved()` are less agnostic of the surroundings. you are right in that there's little difference between the two, though maybe in my head there's something going on saying it's far more likely to want a version _with_ the protected form but _without_ the connected props.

I don't have a big problem with this. I don't think I would do it this way, but I don't see a real problem with it either.